bench/Subscriptions.Benchmark/ConcurrentSubscriptionBenchmark.cs

```diff
@@ -0,0 +1,191 @@
+            {
+                Database = databaseName,
+                Urls = new []{ url },
+                Conventions = new DocumentConventions()
+            };
+            _store.Initialize();
+        }
+
+        public class Thing
+        {
+            public string Name { get; set; }
+        }
+
+        public async Task PerformBenchmark(int workers, int fakeProcessingTimePerBatch, bool script, bool revision)
+        {
+            var runResult = await SingleTestRun(workers, fakeProcessingTimePerBatch, script, revision).ConfigureAwait(false);
+
+            Console.WriteLine("Docs Processed: "+ runResult.DocsProccessed + " ,Docs Requested: " + runResult.DocsRequested + " Elapsed time: " + runResult.ElapsedMs);
+        }
+
+        public void GenerateDocumentsAndRevisions(int amount)
+        {
+            EnableRevisions();
+
+            Console.WriteLine($"Generating {amount} documents..");
+            //generate documents
+            using (BulkInsertOperation bulkInsert = _store.BulkInsert())
+            {
+                for (int i = 0; i <= amount; i++)
+                {
+                    bulkInsert.Store(new Order { Amount = 1 });
+                }
+            }
+        }
+
+        private async Task<RunResult> SingleTestRun(int workers, int fakeProcessingTimePerBatch, bool script, bool revision = false)
+        {
+            try
+            {
+                if (script == false && revision)
+                {
+                    throw new InvalidOperationException("Can't have a revision without script");
+                }
+
+                string revisions = revision ? " (Revisions = true)" : "";
+                if (script)
+                {
+                    SubscriptionCreationOptions subscriptionCreationParams = new SubscriptionCreationOptions
+                    {
+                        Query = "from " + _collectionName + revisions
+                    };
+                    _subscriptionName = await _store.Subscriptions.CreateAsync(subscriptionCreationParams).ConfigureAwait(false);
+                    Console.WriteLine($"Created Subscription with query: '{subscriptionCreationParams.Query}'");
+                }
+                else
+                {
+                    _subscriptionName =  await _store.Subscriptions.CreateAsync<Order>();
+                    Console.WriteLine($"Created Subscription without query");
+                }
+
+                EnableRevisions();
+
+                var workersList = new List<SubscriptionWorker<dynamic>>();
+                var taskList = new List<Task>();
+                Stopwatch sp = null;
+                for (int i = 0; i < workers; i++)
+                {
+                    var worker = _store.Subscriptions.GetSubscriptionWorker(new SubscriptionWorkerOptions(_subscriptionName)
+                    {
+                        Strategy = workers == 1 ? SubscriptionOpeningStrategy.OpenIfFree : SubscriptionOpeningStrategy.Concurrent,
```

Comment:
Let's also test this with a single concurrent worker.

Reply:
Done